src/components/PopoverWithMeasuredContent.js

```diff
@@ -0,0 +1,138 @@
+import _ from 'underscore';
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {View} from 'react-native';
+import Modal from './Modal';
+import {propTypes as modalPropTypes, defaultProps as defaultModalProps} from './Modal/ModalPropTypes';
+import {windowDimensionsPropTypes} from './withWindowDimensions';
+import CONST from '../CONST';
+
+const propTypes = {
+    // All modal props except type, anchorPosition, and the windowDimensions prop types
+    ...(_.omit(modalPropTypes, ['type', 'anchorPosition', ...(_.keys(windowDimensionsPropTypes))])),
+
+    // The horizontal and vertical anchors points for the popover
+    popoverPosition: PropTypes.shape({
+        horizontal: PropTypes.number.isRequired,
+        vertical: PropTypes.number.isRequired,
+    }).isRequired,
+
+    // Where the popover should be positioned relative to the anchor points.
+    anchorOrigin: PropTypes.shape({
+        horizontal: PropTypes.oneOf(_.values(CONST.MODAL.ANCHOR_ORIGIN_HORIZONTAL)),
+        vertical: PropTypes.oneOf(_.values(CONST.MODAL.ANCHOR_ORIGIN_VERTICAL)),
+    }),
+
+    // Optionally pass in a function with content to measure.
+    // Will use this.props.children by default, but if children are not displayed, then the measurement will not work.
+    measureContent: PropTypes.func,
+};
+
+const defaultProps = {
+    // Default modal props
+    ...(_.omit(defaultModalProps, ['type', 'anchorPosition'])),
+
+    // Default positioning of the popover
+    anchorOrigin: {
+        horizontal: CONST.MODAL.ANCHOR_ORIGIN_HORIZONTAL.LEFT,
+        vertical: CONST.MODAL.ANCHOR_ORIGIN_VERTICAL.BOTTOM,
+    },
+
+    // If this prop is null or undefined, then we'll measure the children instead
+    measureContent: null,
+};
+
+class PopoverWithMeasuredContent extends Component {
```

Comment:
Open to a better name for this component. Essentially it's a special popover that measures the content within and then shifts that content based on the `anchorOrigin` props.